agent/proxycfg/state_oss_test.go

```diff
@@ -0,0 +1,343 @@
+// +build !consulent
+
+package proxycfg
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/consul/agent/cache"
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+// This test is meant to exercise the various parts of the cache watching done by the state as
+// well as its management of the ConfigSnapshot
+//
+// This test is expressly not calling Watch which in turn would execute the run function in a go
+// routine. This allows the test to be fully synchronous and deterministic while still being able
+// to validate the logic of most of the watching and state updating.
+//
+// The general strategy here is to
+//
+// 1. Initialize a state with a call to newState + setting some of the extra stuff like the CacheNotifier
+//    We will not be using the CacheNotifier to send notifications but calling handleUpdate ourselves
+// 2. Iterate through a list of verification stages performing validation and updates for each.
+//    a. Ensure that the required watches are in place and validate they are correct
+//    b. Process a bunch of UpdateEvents by calling handleUpdate
+//    c. Validate that the ConfigSnapshot has been updated appropriately
+func TestState_WatchesAndUpdates_OSS(t *testing.T) {
```

Comment:
This is not a new test, it's a copy of the one in `state_test.go` so that the `terminating-gateway-handle-update` case can have diverging implementations.

Reply:
Undoing this in favor of generating the ServiceID string as a variable. That allows the CorrelationIDs in the test case to differ for OSS/ENT.